Google.PowerShell/Common/GcloudWrapper.cs

```diff
@@ -56,43 +51,28 @@ public static class GCloudWrapper
         /// <summary>
         /// Returns the access token of the current active config.
         /// </summary>
-        public static async Task<ActiveUserToken> GetAccessToken()
+        public static async Task<ActiveUserToken> GetAccessToken(CancellationToken cancellationToken)
         {
             // We get the issued time before the command so we won't be too late
             // when it comes to token expiry.
             DateTime issuedTime = DateTime.Now;
 
-            string accessToken = await GetGCloudCommandOutput("auth print-access-token");
-            JToken tokenJson = JObject.Parse(accessToken);
-
-            try
-            {
-                // SelectToken will throw NullReferenceException if the token cannot be found.
-                tokenJson = tokenJson.SelectToken("token_response");
-
-                ActiveUserToken token = tokenJson.ToObject<ActiveUserToken>();
-                token.Issued = issuedTime;
-
-                return token;
-            }
-            catch (NullReferenceException)
-            {
-                // Throw exception at the end.
-            }
-            catch (Newtonsoft.Json.JsonException)
-            {
-                // Throw exception at the end.
-            }
+            string userCredentialJson = await GetGCloudCommandOutput("auth print-access-token");
+            cancellationToken.ThrowIfCancellationRequested();
```

Comment:
I don't think we should wire in the cancellation token if we don't actually honor it. `GetGCloudCommandOutput` will wait until the process exits regardless of the cancellation state. So we should either:

- Be honest with ourselves, and not even add the cancellation token as a parameter.
- Wire the cancellation token down into ProcessUtils where we call `WaitForExit` and kill the process as soon as cancellation is requested. (I did some Googling and you can see an example here https://gist.github.com/mxpv/3798871).

I'm inclined to go with the former, and not bother with wiring in cancellation if we won't actually support it.